### ISSUE TITLE
[Darc/Maestro++] gather-drop does not correctly parse version numbers for some symbol packages

### ISSUE THREAD

**Issue by mmitche** [dotnet/arcade#3957]
Gather-drop was recently augmented to be able to download symbol packages from Azure devops feeds.  This involved parsing out the version number of the package based on the file name. This works fine in almost all cases, except in a few corner cases where we incorrectly identify the symbol package version.  For example:

```
assets/symbols/Microsoft.DotNet.Common.ProjectTemplates.2.2.3.0.0.symbols.nupkg
```

The code in arcade-services to identify version numbers based on asset ID should be augmented to try its best with these kind of things, and we should add tests.

**Comment by mmitche** [comment]
(not required for GA)

**Comment by mmitche** [comment]
Work in progress to fix this for #4203
